### DISCUSSION TITLE
Skywalking Oap Server crashloop

### DISCUSSION THREAD

**Discussion by linshenfanying** [apache/skywalking#12767] [Q&A]
SkyWalking is deployed on Kubernetes with a separated L1 and L2 cluster setup. However, the L1 nodes frequently restart. What could be the reason? How can we avoid this?

We have utilized SkyWalking for application tracing, and the corresponding architecture is shown in the diagram. A total of 220 services and approximately 1000 instances have been integrated. 

The configuration of our OAP Server is 2 CPU cores and 4 GB of memory. We would like to ask for optimization suggestions.

**Comment by wu-sheng** [comment]
2U4G for 1000 instances? I think OAP is already super powerful.

**Reply by wu-sheng** [reply]
APM is a streaming analysis system, that costs resources. You need to assign the proper CPU/Memory to the whole OAP cluster.

**Reply by linshenfanying** [reply]
Multiple instances, a total of 24 instances. There are 18 L1 instances and 6 L2 instances.

**Reply by wu-sheng** [reply]
For large scale, 12c 20g at least per oap node. And setup for oap cluster by following documents.

As I said, OAP is a streaming analysis system, it needs resources, not a simple collector.

**Reply by wu-sheng** [reply]
And make sure, you don't use jdbc(mysql), it can't support large scale.
Choose BanyanDB or Elasticsearch.

**Reply by linshenfanying** [reply]
We are already using Elasticsearch. After changing the configuration of each node to 12C20G, do we need to keep the number of nodes the same?

**Reply by wu-sheng** [reply]
Can't be sure. Just depend on your payload.

**Reply by linshenfanying** [reply]
In the SkyWalking cluster, are there any allocation recommendations for the Receiver and Aggregator? In an architecture with Kafka, which one, Receiver or Aggregator, needs more resources?

**Reply by wu-sheng** [reply]
I don't think you have to set up Receiver and Aggregator if you don't have any specific reason. Just use the mixed role.

**Reply by linshenfanying** [reply]
There are shared documents on the SkyWalking official website that address the same issues we are encountering, so we will use their approach to solve this problem.
